diff --git a/CMakeLists.txt b/CMakeLists.txt
index 709c81f..4c64452 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.12.0)
+cmake_minimum_required(VERSION 3.18.0)
 
 cmake_policy(SET CMP0054 NEW)
 set(SKBUILD_LINK_LIBRARIES_KEYWORD PRIVATE)
@@ -10,8 +10,7 @@ endif()
 
 project(jarowinkler LANGUAGES C CXX)
 
-find_package(PythonExtensions REQUIRED)
-find_package(Python COMPONENTS Interpreter Development)
+find_package(Python COMPONENTS Interpreter Development.Module REQUIRED)
 include(FetchContent)
 
 set(JW_BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
diff --git a/src/jarowinkler/CMakeLists.txt b/src/jarowinkler/CMakeLists.txt
index 8caf71a..62f43c4 100644
--- a/src/jarowinkler/CMakeLists.txt
+++ b/src/jarowinkler/CMakeLists.txt
@@ -11,9 +11,8 @@ function(create_cython_target _name)
 endfunction(create_cython_target)
 
 create_cython_target(_initialize_cpp)
-add_library(_initialize_cpp MODULE ${_initialize_cpp})
+Python_add_library(_initialize_cpp MODULE ${_initialize_cpp})
 target_compile_features(_initialize_cpp PUBLIC cxx_std_14)
 target_include_directories(_initialize_cpp PRIVATE ${RF_CAPI_PATH} ${JW_BASE_DIR}/jarowinkler)
 target_link_libraries(_initialize_cpp PRIVATE jaro_winkler::jaro_winkler)
-python_extension_module(_initialize_cpp)
 install(TARGETS _initialize_cpp LIBRARY DESTINATION src/jarowinkler)
